Tests/Tests/DatastoreCRUD.m

```diff
@@ -1797,6 +1797,20 @@ -(void) testDeleteDocumentUsingIdHasMultipleLeafNodesInTree
     
 }
 
+- (void)testDeleteMutableDocumentRevision
+{
+    CDTMutableDocumentRevision *mutableRev = [CDTMutableDocumentRevision revision];
+    mutableRev.docId = @"aTestDocId";
+    mutableRev.body = @{ @"hello" : @"world" };
+
+    NSError *error;
+    CDTDocumentRevision *rev = [self.datastore createDocumentFromRevision:mutableRev error:&error];
+
+    mutableRev = [rev mutableCopy];
+
+    XCTAssertNotNil([self.datastore deleteDocumentFromRevision:mutableRev error:&error],
+                    @"It should not fail: %@", error);
+}
```

Comment:
Now try to fetch the document from the datastore and assert that it has the deleted flag set